invenio_rdm_records/fixtures/vocabularies.py

```diff
@@ -106,84 +109,249 @@ def __iter__(self):
+        super().__init__(message)
+        self.errors = errors
+
+
+#
+# Fixture
+#
+class VocabulariesFixtures:
+    """Concept of Vocabulary fixtures across locations.
+
+    This class' responsibility is to load different vocabularies in
+    this priority order: app_data, then extensions and then this package.
+
+    Earlier vocabularies in this hierarchy are chosen first. But in the case
+    of subvocabularies (e.g. subject types), having loaded a subvocabulary
+    before, shouldn't prevent loading another one. Only if the same
+    subvocabulary is encountered again should it be ignored.
+
+    Concretely, having loaded MeSH subjects shouldn't prevent loading FAST
+    subjects even though they are both under the "subjects" vocabulary.
+    Another MeSH subject encountered down the hierarchy is ignored however.
+    """
+
+    def __init__(self, identity, app_data_folder=None, pkg_data_folder=None,
+                 filename="vocabularies.yaml", delay=True):
+        """Constructor.
+
+        identity: Identity to use when loading
+        app_data_folder: Path object to instance data folder
+        pkg_data_folder: Path object to this package's data folder.
+                         Defaults to `./data` and really only changeable for
+                         tests.
+        filename: vocabularies filename to check at each location
+        """
         self._identity = identity
+        # Path("./app_data") assumes app_data is in current working directory
+        self._app_data_folder = app_data_folder or Path("./app_data")
+        self._pkg_data_folder = (
+            pkg_data_folder or Path(__file__).parent / "data"
+        )
+        self._filename = filename
+        self._delay = delay
+        # self._ids = set()
 
-    def _open_vocabularies(self):
-        """Open vocabulary file and return its content."""
-        for path in self._search_paths:
-            filepath = path / self._filename
+    def _entry_points(self):
+        """List entrypoints.
 
-            # Providing a vocabularies.yaml file is optional
-            if not filepath.exists():
-                continue
+        Python now officially recommends importlib.metadata
+        (importlib_metadata backport) for entrypoints:
+        - https://docs.python.org/3/library/importlib.metadata.html
+        - https://packaging.python.org/guides/creating-and-discovering-plugins/
+          #using-package-metadata
 
-            with open(filepath) as fp:
-                data = yaml.safe_load(fp) or {}
-                for id_, entry in data.items():
-                    if isinstance(entry["data-file"], dict):
-                        entry["data-file"] = {
-                            k: path / v for k, v in entry["data-file"].items()
-                        }
-                    else:
-                        entry["data-file"] = path / entry["data-file"]
+        but Invenio is much invested in pkg_resources (``entry_points``
```

Comment:
maybe is worth bringing this up so it can be evaluated by architects and potentially worked on later in the future?